core/src/session.rs

```diff
@@ -144,13 +144,15 @@ impl Session {
 
     async fn connect_inner(
         &self,
-        access_point: SocketAddress,
+        access_point: &SocketAddress,
         credentials: Credentials,
     ) -> Result<(Credentials, Transport), Error> {
-        let mut transport = connection::connect(
+        const MAX_RETRIES: u8 = 1;
+        let mut transport = connection::connect_with_retry(
             &access_point.0,
             access_point.1,
             self.config().proxy.as_ref(),
+            MAX_RETRIES,
```

Comment:
One more thing: We never try more than once. Why bother to have `MAX_RETRIES` at all then? In recent testing we saw that sometimes re-trying the same would work wonderfully. Should this value be upped?

Reply:
It's not rust specific. A constant, defined in one place is better than undocumented numbers littered around.  

One retry seems to be enough but this lets us easily tweak it later if required. I'm trying to balance retrying a given AP that's just flakey, whilst also quickly cycling around the other APs in case there's a more severe issue. For example, I've a usecase where port 4070 is blocked, so giving up on that asap and trying 443 is good. I think just once is a reasonable balance but I'm happy to alter it.